### DISCUSSION TITLE
Add Bun pt. 2

### DISCUSSION THREAD

**Discussion by hjonasson** [remix-run/remix#7413] [Proposals]
👋 

[This discussion](https://github.com/remix-run/remix/discussions/7372) suggested adding Bun to the soon-to-be-removed `create-remix` CLI flow. This is a slightly more detailed proposal to support Bun.

I created [this PR](https://github.com/remix-run/remix/pull/7399) to run Remix on Bun. It was pointed out there that in its current state, it does actually spin up a node process alongside the bun processes. I investigated it and found that it is through `node <mylocation>/node_modules/.bin/remix dev`. After experimenting with different `remix.config.js` configurations I find that it seems necessary to create a dedicated `remix-run/bun` package, similar to the Deno one (maybe that was obvious all along 🤷 ?).

As per the [contribution docs](https://github.com/remix-run/remix/blob/main/docs/pages/contributing.md), I decided to start a discussion before spending more time on coding it. I would love to work on it and/or help out if this is accepted depending on my availability.

Thanks!

**Comment by benajaero** [comment]
Hi, has there been any progress on this?

**Reply by sergiodxa** [reply]
The script `remix dev` still can't work with Bun, but you can use Bun as the runtime of your app, just that the build will use Node.

Maybe with the Vite plugin it's possible to make it work.

**Reply by sjdonado** [reply]
It works now with the vite plugin

**Reply by IgnisDa** [reply]
@sjdonado How did you get it working? Do you have a repo I could take a look into?

**Reply by sjdonado** [reply]
> @sjdonado How did you get it working? Do you have a repo I could take a look into?

Yes, feel free to check out https://github.com/sjdonado/remix-dashboard

**Reply by wallpants** [reply]
@sjdonado Your project is not using the Bun runtime. You're only using Bun to manage your dependencies. Try using any Bun specific features in your remix app and you'll see it crash and you'll get an error saying "Bun" is not defined.

**Reply by tmechsner** [reply]
What about this? [https://sergiodxa.com/tutorials/build-an-app-with-remix-and-bun](https://sergiodxa.com/tutorials/build-an-app-with-remix-and-bun)

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
